### PR TITLE
Fixes for PR #14

### PR DESCRIPTION
**Pull request by jordansafer**
Made PR #20 to add logging after adding logging locally, but then saw PR #14 also added logging and was farther along in the process. This is an update to PR #20 with fixes to the readme and using the config file.

### CODE REVIEW

File: plugins/log.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        logFile.write(msg);
+      }
+    }
+  });
+
+  multimeter.addCommand("log", {
+    description: "Log all console entries to a log file",
+    helpText: "Enable/Disable logging to txt file",
+    handler: toggleLogging,
+  });
```
Comment (CGamesPlay) [inline plugins/log.js]: This should be removed. There should only be a single source of truth for logging being disabled/enabled, and I think the config file makes the most sense for that.

File: plugins/log.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const fs = require("fs");
+const logFile = fs.createWriteStream("screeps.log", { flags: "a" });
+const errorLogFile = fs.createWriteStream("screeps.errors.log", { flags: "a" });
```
Comment (CGamesPlay) [inline plugins/log.js]: Can you get the filenames to use from the config file as well? It seems like an obvious place to me, and perhaps someone wants their logs in a "log" subdirectory, for example.

**Review by CGamesPlay — CHANGES_REQUESTED** [review]
Hi @jordansafer, thanks for contributing! i appreciate you taking the time to work on this plugin and apologize for the delay. I'm infrequently at a computer these days 🙂 

Similar to what I wrote in #14, instead of having a `log: true` and `errorLogFile: true`, please use `logFilename: "filename"` and `errorLogFilename: "filename"`. This would allow lots of different functionality:

|`logFilename`|`errorLogFilename`|Effect|
|--|--|--|
| `null` | `null` | No logging.|
| `screeps.log` | `screeps.log` | Logs and errors in one file. |
| `null` | `screeps.log` | Only error messages get logged. |

I would prefer not to have a separate command to enable/disable logging from the program. Imagine this user: sets up `logFile` as per directions, runs `/log disable` because they decide they don't want it, gets frustrated when logs are re-enabled every time they run the program.

On the whole the code looks good, these are hopefully pretty minor changes to make. Thanks again for contributing, and I'll try to respond to your next update more quickly!

File: plugins/log.js (hunk 2 above)
Comment (CGamesPlay) [reply plugins/log.js]: Note here that if both of these happen to be the same filename, the logs would go to the same file. No special code is required for that behavior.

File: plugins/log.js (hunk 1 above)
Comment (jordansafer) [reply plugins/log.js]: Truthfully, I agree, but this was functionality RvstFyth added and it wasn't clear what the direction was so I preserved all the functionality.

File: plugins/log.js (hunk 2 above)
Comment (jordansafer) [reply plugins/log.js]: I agree on this account as well, adding 1 case to the 3 you mentioned above, if errorlogfile is null but normal isn't then all logs go to the same file (as it is currently effectively).

**Comment by jordansafer** [comment]
Updated to remove toggling option and use file names from config

File: plugins/log.js [hunk 3]
```diff
@@ -1,39 +1,19 @@
 const fs = require("fs");
-const logFile = fs.createWriteStream("screeps.log", { flags: "a" });
-const errorLogFile = fs.createWriteStream("screeps.errors.log", { flags: "a" });
 
 module.exports = function(multimeter) {
-  let enabled = multimeter.config.log === "true";
+  let log = multimeter.config.logFilename;
+  if (!log) return;
```
Comment (CGamesPlay) [inline plugins/log.js:5]: What's this for? What if I only want to see errors in my log (which I think is probably the "normal" case for long-running scripts not in active development).

File: plugins/log.js [hunk 4]
```diff
@@ -1,39 +1,19 @@
...
+  let errorLog = log || multimeter.config.errorLogFilename;
```
Comment (CGamesPlay) [inline plugins/log.js:6]: Similarly, this actually prevents you from *disabling* error logs if you want (which, I mean, why would you, but still). A better approach might be:

```javascript
let errorLogFilename =
  typeof multimeter.config.errorLogFilename === 'string'
    ? multimeter.config.errorLogFilename
    : null;

const logFile = log && fs.createWriteStream(...)
const errorLogFile = errorLogFilename && fs.createWriteStream(...)
```

And then of course making sure that the `logFile` and `errorLogFile` variables aren't null before using them.

**Review by CGamesPlay — APPROVED** [review]
I have some more feedback for you, but in the interests of getting this actually shipped I am going to approve now. If you want to file a follow-up PR feel free to.

Thanks for contributing to both @jordansafer and @RvstFyth!

**Pull request merged**